### ISSUE TITLE
Upload Photo to Facebook

### ISSUE THREAD

**Issue by masonry47** [engram-design/SocialPoster#44]
### What are you trying to do?

Hi there!

We are hoping to have the asset from our entry that we want to use in our post actually upload to Facebook.  We see that concept mentioned at the bottom of this [comment](https://github.com/verbb/social-poster/issues/23#issuecomment-474136041). 

### What's your proposed solution?

The reasoning is here is that according to many, Facebook "posts" which have a link attached to them get considerable less "reach".  But... the Facebook API requires a link if providing a picture url:

`(#100) A link must be provided if a picture URL is given.`

So if we were to be actually uploading the photo to Facebook, it seems a link would not need to be provided.  In our use case, the "reach" of the post is really important as it is related to animal welfare (posts about lost and found animals), so needless to say the more people that see it, the better.

Thanks in advance on any thoughts!

**Comment by engram-design** [comment]
That should be possible now, if you ensure that you pick an Assets field to grab the image from. You'll also need to provide a URL for the link, which you'll probably do (likely linking to the entry that you're sharing to socials).

However, I believe there are extended Facebook app permissions you'll need to use to post a photo.

This should also be addressed in the upcoming Social Poster 4 version which features lots of improvements under the hood, and I can confirm posting an image to Facebook works.

**Comment by masonry47** [comment]
Thanks for the info!

The link part is the issue though since we have to provide a link when providing a picture.  So this works...but the photo itself is a link to the URL we provide.  And when a Facebook post has a link to an external website, that is when the post gets less "reach".  So what we are trying to get to... is as if we posted directly on Facebook, where the photo in the post is simply a Facebook hosted photo (when you click on it just opens the photo within Facebooks asset library).  

Thanks a bunch for any more thoughts on it.

**Comment by engram-design** [comment]
Right, so you really want to both upload the image to your Facebook Photos, and use that uploaded photo in your post. That sound correct?

**Comment by masonry47** [comment]
exactly.

**Comment by masonry47** [comment]
Just checking back on the feasibility of this.  Thanks!

**Comment by engram-design** [comment]
Probably not going to be a quick-fix, I'll need to investigate a little further. We've got a major update coming out soon, so should be able to address it after that.
